hack/test-port-forwarding.pl

```diff
@@ -20,8 +20,24 @@
 use Socket qw(inet_ntoa);
 use Sys::Hostname qw(hostname);
 
+# check for nc
+sub check_netcat_installed {
+    my $nc_path = `which nc 2>/dev/null`;
+    chomp $nc_path;
+    unless ($nc_path) {
+        die "Error: 'nc' (netcat) is not installed on the host system.\n" .
+            "Please install netcat to run this test script:\n" .
+            "  - On macOS: brew install netcat\n" .
+            "  - On Ubuntu/Debian: sudo apt-get install netcat\n" .
+            "  - On RHEL/CentOS: sudo yum install nmap-ncat\n";
```

Comment:
It doesn't really matter, but instead of all the explicit newlines and concatenation you could use a here-document:

```suggestion
        die <<"end_of_message";
Error: 'nc' (netcat) is not installed on the host system.
Please install netcat to run this test script:
- On macOS: brew install netcat
- On Ubuntu/Debian: sudo apt-get install netcat
- On RHEL/CentOS: sudo yum install nmap-ncat
end_of_message
```

Reply:
Done 😄